lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -1613,6 +1613,101 @@ export class BrowserCacheManager extends CacheManager {
         this.setInteractionInProgress(false);
     }
 
+    /**
+     * Create request retry key to cache retry status
+     * @param correlationId
+     */
+    generateRequestRetriedKey(correlationId: string): string {
+        return `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REQUEST_RETRY}.${correlationId}`;
+    }
+
+    /**
+     * Gets the request retry value from the cache
+     * @param correlationId
+     */
+    getRequestRetried(correlationId: string): number | null {
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        const cachedRetryNumber = this.getTemporaryCache(requestRetriedKey);
+        if (!cachedRetryNumber) {
+            return null;
+        }
+        return JSON.parse(cachedRetryNumber);
+    }
+
+    /**
+     * Sets the request retry value to "retried" in the cache
+     * @param correlationId
+     */
+    setRequestRetried(correlationId: string): void {
+        this.logger.trace("BrowserCacheManager.setRequestRetried called");
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        this.setTemporaryCache(requestRetriedKey, JSON.stringify(1), false);
+    }
+
+    /**
+     * Removes the request retry value from the cache
+     * @param correlationId
+     */
+    removeRequestRetried(correlationId: string): void {
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        this.removeTemporaryItem(requestRetriedKey);
+    }
+
+    /**
+     * Caches the redirectRequest in the cache
+     * @param redirectRequest
+     */
+    cacheRedirectRequest(redirectRequest: RedirectRequest): void {
+        this.logger.trace("BrowserCacheManager.cacheRedirectRequest called");
+        const { onRedirectNavigate, ...restParams } = redirectRequest;
+        const encodedValue = base64Encode(JSON.stringify(restParams));
```

Comment:
Does this need to be encoded or can it just be stringified?
```suggestion
        const encodedValue = JSON.stringify(restParams);
```

Reply:
I added the encoding to match the existing pattern used for caching the code request, but I suppose that needs it because of the verifier, etc. So this can just be stringified.